src/pages/Library.tsx

```diff
@@ -157,15 +227,316 @@ class Library extends React.Component<ILibraryProps, ILibraryState>
+            }
+    
+        });
+        
+    }
+ 
+    
+    //Scroll to top of current chapter.
+    toTop(){
+        document.querySelector('ion-content')!.scrollToTop(0);
+    }
+
+    //Change the chapter shown to the new chapter, "chapter".
+    setChapter(chapter: any){        
+        //Be sure to scroll to the top of this new chapter.
+        this.toTop();
+        
+        //If we're on the last chapter wrapping back to 1st, be sure to reset the flag for showing special end-banner
+        if(this.state.last && chapter === 10){
+            this.setState(update(this.state, {
+                last : {$set : false},
+                showFab : {$set : false},
+            }));
+        }
+        if(this.state.showFab === true){
+            this.setState(update(this.state, {
+                showFab : {$set : false},
+            }));
+        }
+        console.log("setting context for statement")
         this.setContext(update(this.state.context,{
             Chapter: {statement: {$set:chapter}},
             Section: {saying: {SectionNumber: {$set: chapter}}, statement: {SectionNumber: {$set: chapter}}}, 
         }));
+        
+    }
+
+    //Misnomer. This sets to a Section : Part 
+    setAll(section: any, part: any){
+        this.toTop()
+        console.log("setting context for all")
+            this.setContext(update(this.state.context,{
+                Section: {all: {SectionNumber: {$set: section}, Part: {$set: part}}}, 
+        }));
+        
+    }
+
+
+    //Fetch the prev chapter
+    //Wrapper f(x) for Right+Left nav buttons.
+    prevChapter(){
+        //FOR CLICKING THE PREVIOUS CHAPTER BUTTON
+            
+        //Action dependent upon whether we're viewing the cards or the full chapter.
+        //Turn to chapter curNum
+        if(this.state.context.Mode === "statement"){
+            //Get current chapter#, which is also SectionNumber.
+            var curNum = this.state.context.Chapter.statement;
+            //To compensate for a problem from AllPopoverContent <- which sets Chapter.statement to a STRING instead of an INT 
+            if(typeof curNum === 'string'){
+                curNum = parseInt(curNum);
+            }
+
+            //This is the next button, so we wanna go to next chapter (+ check edge cases)
+            if(curNum === 10){
+                //Do nothing, at the beginning.
+            }else if(curNum === 25){
+                curNum = 22
+            }else{
+                curNum--;
+            }
+
+            //If it's 1st, wrap around to last chapter
+            if(this.state.context.Chapter.statement === 10) {
+                curNum = 29
+            }
+            this.setChapter(curNum);
+        }else{
+            var sec = this.state.context.Section.all.SectionNumber;
+            var pt = this.state.context.Section.all.Part;
+            console.log("sec: " + sec + " pt: " + pt);
+
+            
+            //Sec 0-13: Parts:0
```

Comment:
This is cancer. Great job with the tough logic.

Reply:
T^T thx...